acp/user_profile_rules_avatar_module.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+*
+* User Profile Rules extension for the phpBB Forum Software package.
+*
+* @copyright (c) 2015 phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+*/
+
+namespace phpbb\userprofilerules\acp;
+
+class user_profile_rules_avatar_module
+{
+	/** @var \phpbb\config\db_text */
+	protected $config_text;
+
+	/** @var \phpbb\db\driver\driver_interface */
+	protected $db;
+
+	/** @var \phpbb\log\log */
+	protected $log;
+
+	/** @var \phpbb\request\request */
+	protected $request;
+
+	/** @var \phpbb\template\template */
+	protected $template;
+
+	/** @var \phpbb\user */
+	protected $user;
+
+	/** @var ContainerInterface */
+	protected $phpbb_container;
+
+	/** @var string */
+	protected $phpbb_root_path;
+
+	/** @var string */
+	protected $php_ext;
+
+	/** @var string */
+	public $u_action;
+
+	public function main($id, $mode)
+	{
+		// This is really ugly, we need to do something about this as soon as phpBB enables us to
```

Comment:
Ugly as it is...you're already using $phpbb_container to load some services. What do you think about getting the services from the container instead of using the globals? It may still be ugly, but it's only 1 global instead of 7 ??